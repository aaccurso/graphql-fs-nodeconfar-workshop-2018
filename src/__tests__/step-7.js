const { graphql } = require('graphql');
const fs = require('fs');
const util = require('util');
const { ROOT_PATH } = require('../filesystem');
const schema = require('../schema');

const fsUnlink = util.promisify(fs.unlink);

describe('step 7', () => {
  it('should execute writeFile mutation', async () => {
    const source = `
    mutation writeFile($name: String!, $content: String!) {
      writeFile(name: $name, content: $content) {
        name
      }
    }
    `;
    const variableValues = {
      name: 'test.txt',
      content: 'test',
    };

    expect(await graphql({ schema, source, variableValues })).toMatchSnapshot();
    expect(await fsUnlink(`${ROOT_PATH}/test.txt`)).toBeUndefined();
  });
});
