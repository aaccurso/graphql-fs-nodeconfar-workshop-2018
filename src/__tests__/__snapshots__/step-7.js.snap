// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step 7 should execute writeFile mutation 1`] = `
Object {
  "data": Object {
    "writeFile": Object {
      "name": "test.txt",
    },
  },
}
`;
